_posts/2020-06-23-case-principal-transformer.adoc

```diff
@@ -0,0 +1,134 @@
+(``QUICKSTARTUSER``), and a simple role decoder as follows:
+
+[source]
+----
+/subsystem=elytron/filesystem-realm=exampleRealm:add(path=fs-realm-users, relative-to=jboss.server.config.dir)
+
+/subsystem=elytron/filesystem-realm=exampleRealm:add-identity(identity=QUICKSTARTUSER)
+
+/subsystem=elytron/filesystem-realm=exampleRealm:set-password(clear={password="quickstartPwd1!"}, identity=QUICKSTARTUSER)
+
+/subsystem=elytron/filesystem-realm=exampleRealm:add-identity-attribute(identity=QUICKSTARTUSER, name=Roles, value=["Admin", "Guest"])
+
+/subsystem=elytron/simple-role-decoder=from-roles-attribute:add(attribute=Roles)
+
+----
+
+For more information about creating FileSystem realms along with all of its possible configurations,
+please refer to the https://docs.wildfly.org/20/WildFly_Elytron_Security.html[Elytron documentation].
+
+== Configuring a Case Principal Transformer
+
+Adding a ``case-principal-transformer`` that converts a principal to upper/lower case takes the general form:
+
+[source]
+----
+/subsystem=elytron/case-principal-transformer=transformerName:add(upper-case="...")
+----
+
+In our example, the ``case-principal-transformer`` is configured to specify that a principal should
+be transformed to upper case as follows:
+
+[source]
+----
+/subsystem=elytron/case-principal-transformer=myTransformer:add(upper-case=true)
+----
+
+Note, the ``upper-case`` attribute can be omitted as it will be set to ``true`` by default.
+
+We then configure a new security domain ``caseDomain`` to use our filesystem realm, role decoder
+and the case principal transformer as a ``pre-realm-principal-transformer``:
+
+
+[source]
+----
+/subsystem=elytron/security-domain=caseDomain:add(realms=[{realm=exampleRealm}], \
+default-realm=exampleRealm,permission-mapper=default-permission-mapper, pre-realm-principal-transformer=myTransformer)
+----
+
+Note that creating an additional security domain (``caseDomain``in this case) is not necessary.
+We could alternatively take the default ``ApplicationDomain`` and add the
+FileSystem realm, role-decoder, principal transformer and permission-mapper to it.
+
+We then add our security domain mapping to the EJB3 subsystem, update the ``sasl-authentication-factory`` to use
+our security domain and update the ``http-remoting-connector`` to use our ``sasl-authentication-factory``:
+
+[source]
+----
+/subsystem=ejb3/application-security-domain=other:add(security-domain=caseDomain)
+
+/subsystem=elytron/sasl-authentication-factory=application-sasl-authentication:write-attribute(name=security-domain, value=caseDomain)
+
+/subsystem=remoting/http-connector=http-remoting-connector:write-attribute(name=sasl-authentication-factory,value=application-sasl-authentication)
+----
+
+For more information on the above snippet, please refer to the quickstart this example is based on
+*https://wildfly-security.github.io/wildfly-elytron/blog/advanced-ejb-security/[here]*.
+
+== Configuring the Client (wildfly-config.xml)
```

Comment:
Would be good to mention how to run the client and inspect the output to see that the principal transformer has been used (as mentioned in your quickstart README).

Reply:
I've updated the blog with all the feedback :+1: